### PR TITLE
[BUG-FIX] display customer email if user is not logged in

### PR DESCRIPTION
**Pull request by SirDomin**
| Q               | A
| --------------- | -----
| Branch?         | 1.8?
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

logged out (with same email as registered user):
image

logged in (with same email):
image

### CODE REVIEW

**Review by lchrusciel — COMMENTED** [review]
Can we add 2 scenarios(or even 3) for this behavior? We should cover at least:
1. Guest with registered email
2. Shop user without a full name
3. Shop user with full name

most of the steps seem to be available out-of-the-box. Sample scenario can be found here: [features/checkout/addressing_order/returning_from_addressing_step_to_shop_homepage.feature](https://github.com/Sylius/Sylius/tree/master/features/checkout/addressing_order/returning_from_addressing_step_to_shop_homepage.feature), but I would create a brand new feature file for this.

File: features/checkout/seeing_proper_checkout_name.feature [hunk 1]
```diff
@@ -0,0 +1,30 @@
+@checkout
+Feature: Seeing proper checkout name when im checking out with existing account.
```
Comment (Zales0123) [inline features/checkout/seeing_proper_checkout_name.feature:2]: We should always follow the convention in the feature header:

```
@checkout
Feature: Seeing proper checkout name when I'm checking out with existing account
    In order to ...
    As a ...
    I want to ...
```

File: src/Sylius/Behat/Context/Ui/Shop/Checkout/CheckoutAddressingContext.php [hunk 2]
```diff
@@ -294,6 +294,14 @@ public function iSpecifyThePasswordAs($password)
         $this->addressPage->specifyPassword($password);
     }
 
+    /**
+     * @Then I should see :displayName in checkout header
```
Comment (Zales0123) [inline src/Sylius/Behat/Context/Ui/Shop/Checkout/CheckoutAddressingContext.php:298]: `displayName` -> `purchaserName`?

File: src/Sylius/Behat/Context/Ui/Shop/Checkout/CheckoutAddressingContext.php [hunk 3]
```diff
@@ -294,6 +294,14 @@ public function iSpecifyThePasswordAs($password)
...
+     */
+    public function iShouldSeeInCheckoutHeader(string $displayName): void
+    {
+        Assert::contains($this->selectShippingPage->getPurchaserEmail(), $displayName);
```
Comment (Zales0123) [inline src/Sylius/Behat/Context/Ui/Shop/Checkout/CheckoutAddressingContext.php:302]: As it does not have to be necessary **email**, maybe the method name should be changed? `getPurchaserIdentifier` (or sth similar)?

File: src/Sylius/Behat/Context/Ui/Shop/Checkout/CheckoutAddressingContext.php (hunk 2 above)
Comment (SirDomin) [reply src/Sylius/Behat/Context/Ui/Shop/Checkout/CheckoutAddressingContext.php:298]: thats the problem,  wasnt sure how to name it's email or full name :/

File: features/checkout/seeing_proper_checkout_name.feature [hunk 4]
```diff
@@ -0,0 +1,33 @@
+@checkout
```
Comment (lchrusciel) [inline features/checkout/seeing_proper_checkout_name.feature:1]: File name could be adjusted from `seeing_proper_checkout_name.feature` to `seeing_purchaser_identifier_in_checkout_page.feature`

**Review by lchrusciel — APPROVED** [review]

**Comment by lchrusciel** [comment]
The base of this pull-request was changed, you need fetch and reset your local branch 
if you want to add new commits to this pull request. **Reset before you pull, else commits
may become messed-up.**

Unless you added new commits (to this branch) locally that you did not push yet, 
execute `git fetch origin && git reset "order-checkout-anon"` to update your local branch.
 
Feel free to ask for assistance when you get stuck :+1:

**Comment by lchrusciel** [comment]
The base of this pull-request was changed, you need fetch and reset your local branch 
if you want to add new commits to this pull request. **Reset before you pull, else commits
may become messed-up.**

Unless you added new commits (to this branch) locally that you did not push yet, 
execute `git fetch origin && git reset "order-checkout-anon"` to update your local branch.
 
Feel free to ask for assistance when you get stuck :+1:

**Review by Tomanhez — APPROVED** [review]

**Review by lchrusciel — APPROVED** [review]

**Comment by lchrusciel** [comment]
Thanks, @SirDomin! :1st_place_medal:

**Pull request merged**